server/src/main/java/org/solarbank/server/ServerRestController.java

```diff
@@ -0,0 +1,34 @@
+package org.solarbank.server;
+
+import jakarta.validation.Valid;
+import java.util.Map;
+import org.solarbank.server.dto.UserInputDto;
+import org.solarbank.server.service.CalculateService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api")
+public class ServerRestController {
+
+    @Autowired
+    private CalculateService calculateService;
+
+    /**
+     * runs calculate service layer, parsing user input data.
+     *
+     * @param userInputDto  the user input data from the front end
+     * @return              returns the ResponseEntity object
+     */
+    @PostMapping("/calculate")
+    public ResponseEntity<Map<String, Object>> userInput(
+            @Valid @RequestBody UserInputDto userInputDto) {
+
+        Map<String, Object> result = calculateService.processUserInput(userInputDto);
```

Comment:
Instead of returning the map returned by your method call we should instead have a `CalculateResponse` class which encapsulates the response we want to send back, in the same way you structured your incoming request.

Reply:
added CalculateResult class